scopesim/effects/metis_wcu/metis_wcu.py

```diff
@@ -0,0 +1,442 @@
+    - black-body emission from the black-body source
+    - coupling into the integrating sphere
+    - integrating sphere magnification factor
+    - thermal emission from the integrating sphere
+
+    Configuration file
+    ------------------
+    The default configuration file for the METIS WCU is provided in the irdb.
+    To modify the configuration, copy the file <irdb>/METIS/metis_wcu_config.yaml
+    to the working directory and edit it there. To use the file do e.g.
+    >>> cmds = sim.UserCommands(use_instrument="METIS", set_modes=["wcu_img_lm"]
+    >>> cmds["!WCU.config_file"] = "my_wcu_config.yaml"
+    >>> metis = sim.OpticalTrain(cmds)
+
+    Changing important parameters
+    -----------------------------
+    The temperatures of the black-body source, the integrating sphere as well as the
+    ambient temperature of the WCU (which is the temperature of the focal-plane mask)
+    can be set by calling
+    >>> metis['wcu_source'].set_temperature(bb_temp=1200*u.K, is_temp=320*u.K,
+                                            wcu_temp=295*u.K)
+
+    The flux-controlling masks are currently implemented as a float (in [0, 1]) that
+    gives the fraction of flux transmitted. It can be change with
+    >>> metis['wcu_source'].set_bb_aperture(0.8)
+
+    The focal-plane mask and/or its orientation and position can be changed by calling
+    >>> metis['wcu_source'].set_fpmask("pinhole", angle=10, shift=(0.1, 0))
+
+    To use the lasers instead of the black-body source, do
+    >>> metis['wcu_source'].set_lamp("laser")
+    Which laser is seen depends on the wavelength of the observation. Note that the
+    tunable laser currently cannot be tuned.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        params = {
+            "z_order": [113, 513],
+            "action": "emissivity",
+            "position": 0,  # position in surface table
+        }
+        self.meta.update(params)
+        self.meta.update(kwargs)
+        if 'config_file' in self.meta:
+            config_file = from_currsys(self.meta['config_file'], self.cmds)
+            with open(find_file(config_file)) as fd:
+                config = yaml.safe_load(fd)
+                self.meta.update(config)
+
+        # Check on the presence of one vital parameter
+        if "rho_tube" not in self.meta:
+            raise ValueError(
+                "Parameters not present: please provide config file or parameter values"
+                "for WCUSource")
+
+        self.set_fpmask(maskname=self.meta['current_fpmask'],
+                        angle=self.meta["fpmask_angle"],
+                        shift=self.meta["fpmask_shift"])
+        self._background_source = None
+
+        self.bb_aperture = self.meta['bb_aperture']
+
+        # Define surfaces (the IS surface is currently taken from parent class)
+        # we add another one here (TODO: make this nicer)
+        self.mask_surf = SpectralSurface(cmds = self.cmds)
```

Comment:
```suggestion
        self.mask_surf = SpectralSurface(cmds=self.cmds)
```
Feel free to ignore my nitpicking here 😛 (or add it to a batched commit, but don't bother turning this into a separate commit...)